src/java/org/streamingpool/core/service/impl/TrackKeepingDiscoveryService.java

```diff
@@ -169,6 +166,9 @@ private void checkSameContexOfExecution() {
 
             if (factoryResult.isPresent()) {
                 LOGGER.info("Stream from id '{}' was successfully created by factory '{}'", newId, factory);
+                innerDiscoveryService.idsDiscovered
```

Comment:
The synchronization of this idsDiscovered looks a bit tricky ;-) I think it is correct, but only because you call this method through the synchronousPutIfAbsent() method and because each discovery service has a new HashSet..... This is a bit tricky to understand .... I wonder if one could document this at least somehow...

Reply:
I inlined the method with a comment because it would break the synchronization if by mistake it is called from somewhere else :)